content/webapp/services/prismic/transformers/events.ts

```diff
@@ -1,21 +1,218 @@
 import { link } from './vendored-helpers';
 import { isNotUndefined } from '@weco/common/utils/array';
 import { Query } from '@prismicio/types';
+import {
+  asText,
+  isDocumentLink,
+  parseBoolean,
+  parseFormat,
+  parseGenericFields,
+  parseLabelTypeList,
+  parsePlace,
+  parseSingleLevelGroup,
+  parseTimestamp,
+  parseTitle,
+} from '@weco/common/services/prismic/parsers';
+import { parseSeason } from '@weco/common/services/prismic/seasons';
+import {
+  determineDateRange,
+  determineDisplayTime,
+  getLastEndTime,
+  parseEventBookingType,
+} from '@weco/common/services/prismic/events';
+import { parseEventSeries } from '@weco/common/services/prismic/event-series';
+import { isPast } from '@weco/common/utils/dates';
 
 export function transformEvent(
   document: EventPrismicDocument,
   scheduleQuery?: Query<EventPrismicDocument>
 ): Event {
-  const event: DeprecatedEvent = parseEventDoc(document, scheduleQuery);
+  const data = document.data;
+  const scheduleLength = isDocumentLink(data.schedule.map(s => s.event)[0])
+    ? data.schedule.length
+    : 0;
+  const genericFields = parseGenericFields(document);
+  const eventSchedule =
+    scheduleQuery && scheduleQuery.results
+      ? scheduleQuery.results.map(doc => transformEvent(doc))
+      : [];
+  const interpretations = data.interpretations
+    .map(interpretation =>
+      link(interpretation.interpretationType)
+        ? {
+            interpretationType: {
+              title: parseTitle(interpretation.interpretationType.data?.title),
+              abbreviation: asText(
+                interpretation.interpretationType.data?.abbreviation
+              ),
+              description: interpretation.interpretationType.data?.description,
+              primaryDescription:
+                interpretation.interpretationType.data?.primaryDescription,
+            },
+            isPrimary: Boolean(interpretation.isPrimary),
+            extraInformation: interpretation.extraInformation,
+          }
+        : null
+    )
+    .filter(Boolean);
 
-  return {
-    ...event,
+  const matchedId =
+    data.eventbriteEvent && data.eventbriteEvent.url
+      ? /\/e\/([0-9]+)/.exec(data.eventbriteEvent.url)
+      : null;
+  const eventbriteId =
+    data.eventbriteEvent && matchedId !== null ? matchedId[1] : '';
+
+  const audiences = data.audiences
+    .map(audience =>
+      link(audience.audience)
+        ? {
+            title: asText(audience.audience.data?.title),
+            description: audience.audience.data?.description,
+          }
+        : null
+    )
+    .filter(Boolean);
+
+  const bookingEnquiryTeam: Team | undefined = link(data.bookingEnquiryTeam)
+    ? {
+        id: data.bookingEnquiryTeam.id,
+        title: asText(data.bookingEnquiryTeam.data?.title) || '',
+        email: data.bookingEnquiryTeam.data!.email!,
+        phone: data.bookingEnquiryTeam.data!.phone!,
+        url: data.bookingEnquiryTeam.data!.url!,
```

Comment:
I had to do a bit of coercion here; I think it's all correct, but I assume 500 errors will tell us quickly if it's not.